### ISSUE TITLE
Produce better error message when query plan is impossible

### ISSUE THREAD

**Issue by charithe** [cerbos/cerbos#1192]
For example, given the condition `P.attr.permissions.folder.read.exists(f, R.attr.folder_path.contains(f))`, it's impossible to generate a query plan without knowing the value of `R.attr.folder_path`. This results in the following error message which is not very helpful.

```
error evaluating condition \"P.attr.permissions.folder.read.exists(f, R.attr.folder_path.contains(f))\": no such attribute: id: 4, attributes: [0xc000f2ad20 0xc000f2acc0 0xc000f2ac60]
```

**Comment by haines** [comment]
I don't think this query plan should be impossible. I'd expect to get back an AST like

```
(exists
  P.attr.permissions.folder
  (lambda
    (contains
      R.attr.folder_path
      f)
    f))
```

**Comment by charithe** [comment]
I am torn on this. I agree it's a bug that the planner wasn't able to produce a plan like yours. But, even if it did, could that plan be easily converted into a query?

**Comment by haines** [comment]
Probably not _easily_ but I feel like that is a problem for the consumer of the plan. The query planner should always be able to produce an AST, even if it is complex. It might be that you choose to handle a few special cases like this if it's useful for your policies, rather than trying to solve it in the general case.

**Comment by dbuduev** [comment]
Certainly a bug. Seemingly the query planner expects comprehension targets to be an unknown resource attribute.
Like in this [test](https://github.com/cerbos/cerbos/blob/dab89cf0a2bfd889936cfe15940391153eecb9c9/internal/test/testdata/query_planner/policies/principal_policy_macros.yaml#L44): `R.attr.geos.all(x, !x.countries.exists(t, t in P.attr.forbidden))`. Here's the [AST](https://github.com/cerbos/cerbos/blob/dab89cf0a2bfd889936cfe15940391153eecb9c9/internal/test/testdata/query_planner/suite/macro_user.yaml#L121).

The query plan should provide the AST regardless of the comprehension target. If `P.attr.permissions.folder.read` is `[r1, r2, ..., rN]`. The AST can be as simple as
`(OR (contains folder_path r1) (contains folder_path r2) ... (contains folder_path rN))`.
